sdk/state/state.go

```diff
@@ -37,6 +40,46 @@ type Channel struct {
 	latestUnauthorizedCloseAgreement CloseAgreement
 }
 
+type CloseState int
+
+const (
+	CloseError CloseState = iota - 1
+	CloseNone
+	CloseEarlyClosing // a proposed declTx before fully confirmed is submitted
+	CloseClosing      // latest declTx is submitted
+	CloseNeedsClosing // an earlier declTx is submitted
+	CloseClosed
```

Comment:
Can you use the full name of the type as the prefix?
```suggestion
	CloseStateError CloseState = iota - 1
	CloseStateNone
	CloseStateEarlyClosing // a proposed declTx before fully confirmed is submitted
	CloseStateClosing      // latest declTx is submitted
	CloseStateNeedsClosing // an earlier declTx is submitted
	CloseStateClosed
```

I think we should also reduce this to the states we plan to use now:
```suggestion
	CloseStateNone CloseState = iota - 1
	CloseStateEarlyClosing // a proposed declTx before fully confirmed is submitted
	CloseStateClosing      // latest declTx is submitted
	CloseStateNeedsClosing // an earlier declTx is submitted
	CloseStateClosed
```

Do you think we could get away with a single closing state? If we can, I think we should for now and say that participants always try to close the channel themselves when they see it closing from the other party. We can also say that we always see if we need to copy signatures of declarations submitted too, that way we don't need a state to differentiate that case. It keeps it simpler, probably covers some edge cases automatically by doing so too.
```suggestion
	CloseStateNone CloseState = iota - 1
	CloseStateClosing
	CloseStateClosed
```

Thoughts?

Reply:
replacing `CloseEarlyClosing` with `CloseClosing` and just always trying to grab the declTx sigs I can see. But if the other party submits an earlier declTx, I think that's a special case we'd want to flag differently? eg. in `CloseClosing` the user doesn't need to take action if they don't want, since this is a safe agreement and the other user can close it. 

With `CloseNeedsClosing` the user _needs_ to take action.

wdyt? If we want to simplify for now I can add a TODO for the `CloseNeedsClosing` case

Reply:
also `CloseError` is being used currently, if the EI seq num is an unexpected value for some reason, or there's an error in getting `CloseState()`. We could keep that as `CloseStateNone` but I think an error state will be used pretty soon eg. in [this ticket ](https://github.com/stellar/experimental-payment-channels/issues/218)

Reply:
I'm not sure our statuses can capture every situation whe

I still want to remove `CloseError` and to reassign that scenario to another state. It will be harder to represent CloseError in a general state that is about opening and closing. It sounds like `CloseError` indicates that the channel is broken. We should work really hard to make it so no situation can arise where the channel is broken. I'd argue that there are a finite number of states and we shouldn't have anything outside them:

- Not open - Seq < s means not open yet
- Open - Seq = s, formation tx ingested and successful, open
- Closing - Seq > s, signers still engaged, closing
- Closed - Seq > s, signers reverted, closed

I'm simplifying a bit for where there has been no withdrawal, which introduces `e`.

I feel like for the open case we want to be absolutely sure that the formation tx succeeded, but in the closing/closed states it's more about the seq num and the state of signers on the users escrow account (not both escrow accounts).

For the close states though we can use sequence to indicate if there is a tx submitted, then should evaluate the state of the account, which we can do with the tx meta and result data. Using only the sequence might be misleading. For example if I close earlier 

Then if we need it add something for figuring out what state the channel is closing in, but that might not even be a state. Or make it easier to participate in the close whichever case. I feel like if we can do the latter it will result in less code to maintain.

Reply:
removed the `CloseStateError` here
https://github.com/stellar/experimental-payment-channels/pull/232/commits/ba856ea6f2a6625d804e92eef3cce67a8e640d29

if alright with you I'd like to push this PR through, and then refactor the CloseState with this ticket https://github.com/stellar/experimental-payment-channels/issues/218 : which will be incorporating the Open states, and taking into account the tx meta and result data as your comment talks about.

wdyt?